### PR TITLE
Placeholders do not work when downloading from virtual repo

### PR DESCRIPTION
**Pull request by DimaNevelev**

### CODE REVIEW

File: artifactory/services/movecopy.go [hunk 1]
```diff
@@ -109,7 +108,9 @@ func (mc *MoveCopyService) moveFiles(regexpPath string, resultItems []utils.Resu
 				destPathLocal = clientutils.TrimPath(destPathLocal + "/" + v.Path + "/")
 			}
 		}
-		destFile, err := clientutils.ReformatRegexp(regexpPath, v.GetItemRelativePath(), destPathLocal)
+		givenPath := clientutils.CleanRepoFromPath(params.GetFile().Pattern)
+		actualPath := clientutils.CleanRepoFromPath(v.GetItemRelativePath())
+		destFile, err := clientutils.ReformatTargetByPaths(givenPath, actualPath, destPathLocal)
```
Comment (eyalbe4) [inline artifactory/services/movecopy.go]: The above 3 lines seem to be duplicated (see the *download.go*).
I suggest moving them into *utils.go* (after consolidating them into one function).

File: utils/utils_test.go [hunk 2]
```diff
@@ -3,18 +3,34 @@ package utils
...
 
-func TestReformatRegexp(t *testing.T) {
-	assertReformatRegexp("1(.*)234", "1hello234", "{1}", "hello", t)
-	assertReformatRegexp("1234", "1hello234", "{1}", "{1}", t)
-	assertReformatRegexp("1(2.*5)6", "123456", "{1}", "2345", t)
-	assertReformatRegexp("(.*) somthing", "doing somthing", "{1} somthing else", "doing somthing else", t)
-	assertReformatRegexp("(switch) (this)", "switch this", "{2} {1}", "this switch", t)
-	assertReformatRegexp("before(.*)middle(.*)after", "before123middle456after", "{2}{1}{2}", "456123456", t)
-	assertReformatRegexp("", "nothing should change", "nothing should change", "nothing should change", t)
+func TestCleanRepoFromPath(t *testing.T) {
```
Comment (eyalbe4) [inline utils/utils_test.go]: Let's rename *TestCleanRepoFromPath* to *TestRemoveRepoFromPath*

**Comment by eyalbe4** [comment]
@DimaNevelev, see my inline comments.

File: artifactory/services/download.go [hunk 3]
```diff
@@ -347,8 +347,9 @@ func (ds *DownloadService) createFileHandlerFunc(buildDependencies [][]utils.Fil
 				return nil
 			}
 
-			regexpPattern := clientutils.PathToRegExp(downloadData.DownloadPath)
-			placeHolderTarget, e := clientutils.ReformatRegexp(regexpPattern, downloadData.Dependency.GetItemRelativePath(), downloadData.Target)
+			givenPath := clientutils.CleanRepoFromPath(downloadData.DownloadPath)
```
[the remark this answers is not in the document]
Comment (DimaNevelev) [reply artifactory/services/download.go]: What do you think about renaming it to "pattern"?

File: artifactory/services/download.go [hunk 4]
```diff
@@ -347,8 +347,9 @@ func (ds *DownloadService) createFileHandlerFunc(buildDependencies [][]utils.Fil
...
+			actualPath := clientutils.CleanRepoFromPath(downloadData.Dependency.GetItemRelativePath())
```
[the remark this answers is not in the document]
Comment (DimaNevelev) [reply artifactory/services/download.go]: What do you think about renaming it to "path"?

**Comment by DimaNevelev** [comment]
> @DimaNevelev, see my inline comments.

@eyalbe4, All done. 
Please see the updates.

**Pull request merged**